# Exercise 1
# Step 1.1
The GC count per read and per base sequence content stand out as problematic. The former does not match the the theoretical distribution for both samples. The latter is problematic at the beginning of reads. This is case possibly because the data are RNA-seq samples derived from Drosophila melanogaster midguts specifically and not from other tissues.

# Step 1.2
The origin of the most overrepresented sequence in this sample is from Drosophila melanogaster strain Canton S serine protease 3, serine protease 1, and serine protease 2 genes. Since serine proteases and their homologs are the second largest family of genes in the Drosophila melanogaster genome, it does make sense that they are the most overrepresented sequence. 

# Exercise 2
If I were to reject any samples with the percentage of unique reads less than 45%, I would keep 15 samples. I can see the blocks of triplicates clearly, and this means that the consistency between replicates is high. 

# Exercise 3
# Step 3.3
LFC-Fe Replicate 3 and Fe Replicate 1 have the wrong labels.

# Step 3.6
The categories of enrichments make sense beacause many of these enriched GO terms are associted with developmental processes, gamete generation, and metabolic processes in Drosophila melanogaster.